### PR TITLE
Remove meaningless comment decoration from code

### LINKED ISSUE #62601: Meaningless usage of comment decorators
We have a number of places in the code base that use 'comment decorators' i.e. from `rustlib/src/rust/src/libcore/option.rs`
```
/////////////////////////////////////////////////////////////////////////////
// Type implementation
/////////////////////////////////////////////////////////////////////////////

impl<T> Option<T> {
```
This adds no extra meaning to the code and is totally specific to the developer that wrote it.  We should remove this sort of thing :)
```
[path/to/rustlib/src/rust/src/libcore/] $ egrep -R '/////////////////////' | wc -l
74
```

This issue has been assigned to @sd234678 via [this comment](https://github.com/rust-lang/rust/issues/62601#issuecomment-518989825).

### PR DESCRIPTION
**Pull request by sd234678**
Closes #62601

### CODE REVIEW

**Comment by sd234678** [comment]
r? Centril

**Comment by petrochenkov** [comment]
They are not necessarily meaningless, they may partition code into some logical parts, and highlight those parts and comments applying to those whole parts visually.

It doesn't necessarily have to be `///////////////////////` (which are unintentionally treated as doc comments), just something big and visible "from the distance".

**Comment by sd234678** [comment]
They're not treated as doc comments are they? Some of the ones I left in were trying to illustrate it such as the ones in `src/test/ui/issues/issue-10638.rs`.
The majority of the deleted ones were the only comments in the file, so most people's syntax highlighting will make them stand out. They also differ from the doc comments, so they stand out that way. Some of them use capitalisation to stand out, and looking through as I removed them none of them seemed to have their value diminished by the lack of separators.
I guess the discussion is more whether the issue should be there in the first place, but no one has suggested otherwise on it.

**Comment by sd234678** [comment]
Further to this, if code within a file is in so many distinct parts that it needs partitioning, perhaps it shouldn't be in the same module?

**Comment by sd234678** [comment]
Hmm, I'm not really sure what's causing the errors here. The only diff from master for the file with problems is comment removal.

File: src/grammar/parser-lalr.y [hunk 1]
```diff
@@ -202,9 +202,7 @@ extern char *yytext;
 
 %%
 
-////////////////////////////////////////////////////////////////////////
```
Comment (Centril) [inline src/grammar/parser-lalr.y]: Unnecessary change; the `src/grammar/` directory is going away sooner or later.

File: src/test/pretty/stmt_expr_attributes.rs [hunk 2]
```diff
@@ -259,8 +259,6 @@ fn _12() {
     }
 }
 
-/////////////////
```
Comment (Centril) [inline src/test/pretty/stmt_expr_attributes.rs]: The changes in `src/test/` are all good; can you `cherry-pick` those out to a separate PR for fast-tracking?

File: src/libcore/hash/mod.rs [hunk 3]
```diff
@@ -538,8 +538,6 @@ impl<H> PartialEq for BuildHasherDefault<H> {
 #[stable(since = "1.29.0", feature = "build_hasher_eq")]
 impl<H> Eq for BuildHasherDefault<H> {}
 
-//////////////////////////////////////////////////////////////////////////////
```
Comment (Centril) [inline src/libcore/hash/mod.rs]: This is also good; comment is useless.

**Review by Centril — COMMENTED** [review]
Overall I'm conflicted about removing some of these section comments; they seem useful and I use them myself from time to time (with `// =====`). In other words, I basically agree with @petrochenkov. That said, perhaps the better thing to do here is to look for opportunities to split modules. I would look for the largest sections which fit as a coherent unit and split those out.

**Comment by petrochenkov** [comment]
>Further to this, if code within a file is in so many distinct parts that it needs partitioning, perhaps it shouldn't be in the same module?

No, not at all, the sections may be small and contain just several logically related methods.
This is really a matter of personal taste.

**Comment by petrochenkov** [comment]
Please, keep the highlighting in `librustc_privacy/lib.rs` (in some form) and don't split it into modules, otherwise I don't care much.

**Comment by Centril** [comment]
> > 
> 
> No, not at all, the sections may be small and contain just several logically related methods.
> This is really a matter of personal taste.

Personally I think it largely depends on size; if a section is > 500 LOC and makes sense as a logical unit then a module is probably a good thing.

File: src/test/pretty/stmt_expr_attributes.rs (hunk 2 above)
Comment (sd234678) [reply src/test/pretty/stmt_expr_attributes.rs]: I've done this in #63411

**Comment by sd234678** [comment]
I've put back the ones in `librustc_privacy/lib.rs`. I agree they make it more readable but feel that's probably because something needs splitting out given that it's almost 2000 LOC total. Alas, I'm not quite sure what that would be.

**Comment by sd234678** [comment]
@Centril I've reviewed each file in this and tried to maintain decoration only where the comments already stand out sufficiently in the file.

I've also removed the decoration where unneeded in comments such as 
```rust
// ----- comment -------------
```

Does anyone have any strong opinions on any of these files?

**Review by Centril — COMMENTED** [review]
The changes in `src/test` look good aside from the comments below. Can you extract those to a separate PR including with the comments I've left?

As for the other changes in the code, I continue to feel conflicted about doing the changes there.
It's not clear to me that it is a win and perhaps it's best to just not rock the boat?

**Review by nikomatsakis — COMMENTED** [review]
I appreciate the desire to clean things up. That said, personally, I'm inclined to leave the section headers. Some of them may be superfluous but overall I find them helpful for giving some "structure" to the files. Perhaps that's because I added a lot of them, though. =)

I guess overall it's not something I have a super strong opinion about, but it seems like all of us reviewers are kind of "slightly negative", which makes me think it's not worth the trouble to make the change.

**Comment by sd234678** [comment]
Thanks @nikomatsakis. With the ones that've been extracted into other PRs as suggested by Centril a fair amount of the superfluous ones have been removed so I'm happy for this to be closed if you think that's enough.

**Comment by nikomatsakis** [comment]
@sd234678 thanks for understanding. Closing.

**Pull request closed without merging**